src/cpp/src/block_manager.hpp

```diff
@@ -480,7 +494,7 @@ class BlockManager {
             if (block != nullptr) {
                 block->set_timestamp(std::chrono::system_clock::now());
                 m_block_table[seq_id].push_back(block);
-                group->update_processed_tokens_num(content_len);
+                group->update_processed_tokens_num(content_len == prompt_ids.size() ? content_len - 1 : content_len);
```

Comment:
why do we need such condition?

Reply:
At the end of each `step()` we generate one token and append it to sequences in group.
In case when two identical prompts are used twice, we fully restore kv-cache for sequence group and have prompt length == content_length, but the token which should be generated in the end of `step()` in `generated_ids` is not restored and lost.
So this condition is needed to prevent loosing of this token and run one `step()` to generate it.

Reply:
ok, it would be great to have such explanation as part of the code.